helix-lsp/src/client.rs

```diff
@@ -1114,6 +1115,31 @@ impl Client {
         Some(self.call::<lsp::request::InlayHintRequest>(params))
     }
 
+    pub fn text_document_color_swatches(
+        &self,
+        text_document: lsp::TextDocumentIdentifier,
+        work_done_token: Option<lsp::ProgressToken>,
+    ) -> Option<impl Future<Output = Result<Value>>> {
+        let capabilities = self.capabilities.get().unwrap();
+
+        match capabilities.color_provider {
```

Comment:
Maybe merge with the above:

```rust
if self.capabilities.get().map(|c| c.color_provider).is_none() {
    return None;
}
```

Reply:
See https://github.com/helix-editor/helix/pull/12308#discussion_r1894095036

Reply:
Here it should be fine to just merge the two. It just makes the code more concise

Reply:
> Here it should be fine to just merge the two. It just makes the code more concise

ok, I merged it!